### PR TITLE
internals refactoring & rename DeviceMonitNew to EnsureDeviceApp

### PR DESCRIPTION
**Pull request by gre**
also drop DeviceMonit
prepare future for Ethereum impl

### CODE REVIEW

**Comment by gre** [comment]
( this basically integrate the latest lib too ) cc @meriadec

File: src/internals/devices/ensureDeviceApp.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    ...options
+  }: {
+    currencyId: string,
+    devicePath: string,
+    accountPath: ?string,
+    accountAddress: ?string,
+  },
+) => {
+  try {
+    invariant(currencyId, 'currencyId is defined')
```
Comment (meriadec) [inline src/internals/devices/ensureDeviceApp.js]: `is not defined` ?

File: src/internals/devices/ensureDeviceApp.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// @flow
+
+import invariant from 'invariant'
+import CommNodeHid from '@ledgerhq/hw-transport-node-hid'
+import type Transport from '@ledgerhq/hw-transport'
+import type { IPCSend } from 'types/electron'
+import getAdressFromCurrency from './getAddressForCurrency'
```
Comment (meriadec) [inline src/internals/devices/ensureDeviceApp.js]: `getAddressFromCurrency`

File: src/internals/devices/ensureDeviceApp.js (hunk 1 above)
Comment (gre) [reply src/internals/devices/ensureDeviceApp.js]: It's unclear to me if the comment here should be "what should happen" (an "invariant") or the message when it fails

Comment (gre) [reply src/internals/devices/ensureDeviceApp.js]: yeah i think you are correct, i'll fix it.

File: src/internals/devices/ensureDeviceApp.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ import invariant from 'invariant'
 import CommNodeHid from '@ledgerhq/hw-transport-node-hid'
 import type Transport from '@ledgerhq/hw-transport'
 import type { IPCSend } from 'types/electron'
-import getAdressFromCurrency from './getAddressForCurrency'
+import getAdressForCurrency from './getAddressForCurrency'
```
Comment (meriadec) [inline src/internals/devices/ensureDeviceApp.js]: sry, was also referring about the `dd` of `Address` :D

**Pull request merged**